File: src/wfrp-helper-web/src/model/crafting/Inventory.js
import Item from "./Item";
import Entity from "../Entity";

export default class Inventory extends Entity {
    item: Item;
    amount: number = 1;

    updateWith(entity: Inventory): Inventory {
        super.updateWith(entity);
        this.updateEntityProp(entity, 'item', () => new Item());
        return this;
    }
}

File: src/main/java/pl/khuzzuk/wfrp/helper/model/crafting/inventory/Inventory.java
package pl.khuzzuk.wfrp.helper.model.crafting.inventory;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.envers.Audited;
import org.hibernate.envers.RelationTargetAuditMode;
import pl.javahello.DTO;
import pl.khuzzuk.wfrp.helper.repo.BaseEntity;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Getter
@Setter
@Entity
@Table(schema = "creature")
@Audited
@DTO
public class Inventory extends BaseEntity {
    @ManyToOne
    @JoinColumn
    @Audited(targetAuditMode = RelationTargetAuditMode.NOT_AUDITED)
    private MiscItem item;
    private float amount;
}
